aggregate.go

```diff
@@ -2,13 +2,56 @@ package versionbundle
 
 import (
 	"fmt"
-	"reflect"
 	"sort"
 
 	"github.com/giantswarm/microerror"
 	"github.com/giantswarm/micrologger"
 )
 
+/*
+Core design behind Aggregate() implementation:
+
+Aggregate() function takes list of bundles and it builds all possible
+combinations of them. Only restrictions are possibly conflicting Bundle
+dependencies.
+
+Aggregate() implementation works in three phases:
+
+Assume input [simplified] bundles:
+  []Bundle{ (A, 1), (A, 2), (B, 1), (B, 2), (B, 3), (C, 1), (C, 2)}
+
+1. Group all bundles by name and remove duplicate versions.
+
+This produces following map:
+  {
+	  "A": [(A, 1), (A, 2)]
+	  "B": [(B, 1), (B, 2), (B, 3)]
+	  "C": [(C, 1), (C, 2)]
+  }
+
+2. Build a layered tree from Bundles
+
+                                        (root)
+                                          |
+                    +---------------------+---------------------+
+                    |                                           |
+                    |                                           |
+                  (A,1)                                       (A,2)
+                    |                                           |
+      +-------------+-------------+               +-------------+-------------+
+      |             |             |               |             |             |
+    (B,1)         (B,2)         (B,3)           (B,1)         (B,2)         (B,3)
+      |             |             |               |             |             |
+  +---+---+     +---+---+     +---+---+       +---+---+     +---+---+     +---+---+
+  |       |     |       |     |       |       |       |     |       |     |       |
+(C,1)   (C,2) (C,1)   (C,2) (C,1)   (C,2)   (C,1)   (C,2) (C,1)   (C,2) (C,1)   (C,2)
+
+
+3. Walk the tree and create aggregated bundles where are no dependency
+   conflicts.
+
+*/
```

Comment:
I think it would make sense to have this as part of the godoc of the `Aggregate` method.

Reply:
I'm not fully sure if this is useful information for the API user? I meant this more for the maintenance of this code so that it would be easy to understand the logic how `Aggregate` works.

Reply:
However you like.